src/estimate.c

```diff
@@ -44,15 +45,38 @@ estimate_max_spread_var(PlannerInfo *root, Var *var)
 	if (!valid)
 		return INVALID_ESTIMATE;
 
+	/* We create a subtransaction to run the functions below, in case they
+	 * abort. */
+	BeginInternalSubTransaction(NULL);
+
 	PG_TRY();
 	{
 		max = ts_time_value_to_internal(max_datum, var->vartype);
 		min = ts_time_value_to_internal(min_datum, var->vartype);
+		ReleaseCurrentSubTransaction();
 	}
 	PG_CATCH();
 	{
-		valid = false;
+		MemoryContext ecxt = MemoryContextSwitchTo(ccxt);
+		ErrorData *edata = CopyErrorData();
```

Comment:
Why do you need to copy error data here. Isn't ErrorData  valid until you call FlushErrorState? I don't see you using ErrorData after FlushErrorState, so copying it seems unnecessary. Am I missing something?

Reply:
Maybe I am missing something, but both `errordata` (the error stack) and `errordata_stack_depth` are internal to `elog.c` and there is no other external reference to the TOS and no functions that can be used to read the TOS except `CopyErrorData`, so AFAICT, there is no way to directly reference the TOS.

Reply:
I missed that there is a function `geterrcode()` that does get the error code for TOS.

Reply:
Passes all tests, so updated to use this function instead.